runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/RegisterTask.java

```diff
@@ -305,9 +258,11 @@ else if ("http".equals(configURL.getProtocol()) || "https".equals(configURL.getP
         {
             errors.forEach(msg -> errorHandler.onError(new JsonException(msg)));
         }
-
-        // TODO: repeat to detect and apply changes
-
         return null;
     }
+
+    public void setConfigText(String configText)
+    {
+        this.configText = configText;
+    }
```

Comment:
As discussed, in the Engine, we should track `configURL` to parsed `NamespaceConfig` which may be initially `null` before the first parse of a new `configURL`.

When the engine detects a config change from the watcher task, it can trigger the unregister of the old namespace and a parse plus register of the the new config.